api/src/test/kotlin/edu/wgu/osmt/keyword/KeywordControllerTest.kt

```diff
@@ -123,4 +127,43 @@ internal class KeywordControllerTest @Autowired constructor(
         Assertions.assertThat(result.body!!.framework).isEqualTo("updated framework")
         Assertions.assertThat((result as ResponseEntity).statusCode).isEqualTo(HttpStatus.OK)
     }
-}
\ No newline at end of file
+
+    @Test
+    fun `searchRelatedSkills() should retrieve an existing RSD`() {
+        // arrange
+        val category1 = keywordRepository.create(KeywordTypeEnum.Category, "category1")
+        val category2 = keywordRepository.create(KeywordTypeEnum.Category, "category2")
+
+        richSkillRepository.create(
+            user = "user1",
+            updateObject = RsdUpdateObject(
+                name = "skill-1",
+                statement = "Skill 1",
+                keywords = ListFieldUpdate(add = listOf(category1!!, category2!!)),
+            ),
+        )
+
+        richSkillRepository.create(
+            user = "user1",
+            updateObject = RsdUpdateObject(
+                name = "skill-2",
+                statement = "Skill 2",
+                keywords = ListFieldUpdate(add = listOf(category1!!)),
+            ),
+        )
+
+        // act
+        val result = kwController.searchCategorySkills(
+            uriComponentsBuilder = UriComponentsBuilder.newInstance(),
+            id = category2.id.value,
+            size = 10,
+            from = 0,
+            sort = SkillSortEnum.defaultSort.toString(),
+            status = arrayOf(PublishStatus.Published.toString()),
+        )
+
+        // assert
+        Assertions.assertThat(result).isNotNull
+        Assertions.assertThat(result.body?.size == 1)
```

Comment:
we should add an assertion (and the parameter to use) to check on the result when searching for a specific ApiSearch request body